modules/sfp_dnsdb.py

```diff
@@ -0,0 +1,333 @@
+        },
+    }
+
+    opts = {
+        "api_key": "",
+        "age_limit_days": 0,
+        "verify": True,
+        "cohostsamedomain": False,
+        "maxcohost": 100,
+    }
+
+    optdescs = {
+        "api_key": "DNSDB API Key.",
+        "age_limit_days": "Ignore any DNSDB records older than this many days. 0 = unlimited.",
+        "verify": "Verify co-hosts are valid by checking if they still resolve to the shared IP.",
+        "cohostsamedomain": "Treat co-hosted sites on the same target domain as co-hosting?",
+        "maxcohost": "Stop reporting co-hosted sites after this many are found, as it would likely indicate web hosting.",
+    }
+
+    results = None
+    errorState = False
+    cohostcount = 0
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+        self.cohostcount = 0
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    def watchedEvents(self):
+        return ["IP_ADDRESS", "IPV6_ADDRESS", "DOMAIN_NAME"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return [
+            "RAW_RIR_DATA",
+            "INTERNET_NAME",
+            "INTERNET_NAME_UNRESOLVED",
+            "PROVIDER_DNS",
+            "DNS_TEXT",
+            "PROVIDER_MAIL",
+            "IP_ADDRESS",
+            "IPV6_ADDRESS",
+        ]
+
+    def query(self, endpoint, queryType, query):
+        if endpoint not in ("rrset", "rdata"):
+            self.sf.error(
+                f"Endpoint MUST be rrset or rdata, you sent {endpoint}", False
+            )
+            return None
+
+        if queryType not in ("name", "ip"):
+            self.sf.error(f"Query type MUST be name or ip, you sent {queryType}", False)
+            return None
+
+        headers = {"Accept": "application/x-ndjson", "X-API-Key": self.opts["api_key"]}
+
+        res = self.sf.fetchUrl(
+            f"https://api.dnsdb.info/dnsdb/v2/lookup/{endpoint}/{queryType}/{query}",
+            timeout=self.opts["_fetchtimeout"],
+            useragent="SpiderFoot",
+            headers=headers,
+        )
+
+        if res["code"] == "429":
+            self.sf.error("You are being rate-limited by DNSDB", False)
+            self.errorState = True
+            return None
+
+        if res["content"] is None:
+            self.sf.info(f"No DNSDB record found for {query}")
+            return None
+
+        splittedContent = res["content"].strip().split("\n")
+        if len(splittedContent) == 2:
+            self.sf.info(f"No DNSDB record found for {query}")
+            return None
+
+        try:
+            records = []
+            for content in splittedContent:
+                records.append(json.loads(content))
+        except Exception as e:
+            self.sf.error(f"Error processing JSON response from DNSDB: {e}", False)
+            return None
+
+        return records[1:-1]
```

Comment:
Any likelihood of an out of bounds error here if the length of `splittedContent` isn't as expected?

Reply:
I fixed this by checking is`splittedContent` less or equal to two, if it is it will return None, else it will return records